src/target/riscv/riscv-013.c

```diff
@@ -5367,6 +5368,14 @@ static enum riscv_halt_reason riscv013_halt_reason(struct target *target)
 		return RISCV_HALT_INTERRUPT;
 	case CSR_DCSR_CAUSE_GROUP:
 		return RISCV_HALT_GROUP;
+	case CSR_DCSR_CAUSE_OTHER:
+		if (get_field(dcsr, CSR_DCSR_EXTCAUSE) == 0) {
+			LOG_TARGET_INFO(target, "halted because of hart in a critical error state.");
+			return RISCV_HALT_CRITERR;
+		}
+		LOG_TARGET_ERROR(target, "Unknown DCSR extcause field: 0x%"
+				PRIx64, get_field(dcsr, CSR_DCSR_EXTCAUSE));
+		return RISCV_HALT_UNKNOWN;
```

Comment:
If you like, this could be a switch instead of if cascade.

This will encourage better coding style in future, when there are more EXTCAUSE values.

```suggestion
		switch (get_field(dcsr, CSR_DCSR_EXTCAUSE)) 
		{
		case 0: 
			/* ... */
			return RISCV_HALT_CRITICAL_ERROR;
		default:
			/* ... */
			return RISCV_HALT_UNKNOWN;
		}
```

Reply:
Addressed, thanks.